### PR TITLE
Change debugging from ARGO macros to dynamic debug statements

### PR DESCRIPTION
**Pull request by nkrasnoff**
Previously debugging for the argo module was done via macros that enabled debug statements in the module
at compile time. This PR replaces these macros with dynamic debug statements that can be enabled and disabled
at will at runtime.

Dynamic Debugging usage: https://www.kernel.org/doc/html/latest/admin-guide/dynamic-debug-howto.html

### CODE REVIEW

File: argo-linux/argo-module.c [hunk 1]
```diff
@@ -2364,7 +2364,16 @@ argo_send_stream(struct argo_private *p, const void *_buf, int len,
         iovs[0].pad = 0;
         iovs[1].pad = 0;
 
-        print_hex_dump_bytes("stream send dump: ", DUMP_PREFIX_OFFSET, (void *) buf, to_send);
+       	/* As argo has no need to access data it is sending copy_from_user is
```
Comment (jandryuk) [inline argo-linux/argo-module.c]: The formatting is strange here and is fixed in the last commit.  You should just squash the fix in here.

**Comment by eric-ch** [comment]
Build [$10128](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10128)

File: argo-linux/argo-module.c [hunk 2]
```diff
@@ -3185,100 +3185,136 @@ argo_ioctl(struct file *f, unsigned int cmd, unsigned long arg)
...
 
-    pr_debug("argo_ioctl cmd=%x pid=%d\n", cmd, current->pid);
-
     if (_IOC_TYPE (cmd) != ARGO_TYPE)
         return rc;
 
     switch (cmd)
     {
         case ARGOIOCSETRINGSIZE:
+        pr_debug("Start: SETRINGSIZE=%x pid=%d\n", cmd, current->pid);
```
Comment (dpsmith) [inline argo-linux/argo-module.c:3194]: Here and on the other cases please don't put the pr_* statements outside the opening bracket. I understand you are doing this to deal with the error about variable declarations but the pr_* statement doesn't have to be the absolute first message. You can do the open bracket, variable declaration, pr_* statement, then the remainder of the logic for the case.

**Comment by dpsmith** [comment]
Over all the changes are really good but a major comment is to be careful with scope creep. Clearly you started with trying to clean up the logging macros and end up replacing all the spinlock and memory allocation logic. Granted, I have pushed that we drop all this custom code and use proper kernel functions with this being a very good job at doing that. With that said, I would prefer the commits be ordered in a more cohesive manner. Below is the ordering that I would suggest,

1. Replace live printks with appropriate pr_* macros 
2. Remove ARGO_TRACE calls
3. Replace hexdumps and summary_ring with Dynamic macros
4. Update logging messages and rework ioctl logging
5. Remove argo_* locking macros
6. Remove argo_memory macros
7. Fix crash from SMAP violation

**Comment by nkrasnoff** [comment]
> Over all the changes are really good but a major comment is to be careful with scope creep. Clearly you started with trying to clean up the logging macros and end up replacing all the spinlock and memory allocation logic.

The logic actually doesn't change on non debug builds. Previously there was custom code that was only used when the DEBUG flag was set. Otherwise the macros resulted in using the standard logic as it appears now. What I did was just replace the macros with the code used when DEBUG was not set.

**Comment by dpsmith** [comment]
> > Over all the changes are really good but a major comment is to be careful with scope creep. Clearly you started with trying to clean up the logging macros and end up replacing all the spinlock and memory allocation logic.
> 
> The logic actually doesn't change on non debug builds. Previously there was custom code that was only used when the DEBUG flag was set. Otherwise the macros resulted in using the standard logic as it appears now. What I did was just replace the macros with the code used when ARGO_DEBUG/ARGO_DEBUG_LOCKS was not set.

Right and so you changed how it works when building in DEBUG, although you now have made it work like non-DEBUG which is a good thing.

**Comment by eric-ch** [comment]
Build [$10133](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10133)

File: argo-linux/argo-module.c [hunk 3]
```diff
@@ -3580,30 +3660,40 @@ argo_ioctl(struct file *f, unsigned int cmd, unsigned long arg)
...
                               sizeof(struct xen_argo_viptables_list)) )
-              return -EFAULT;
+            {
+                rc = -EFAULT;
+                break;
+            }
         }
         break;
         default:
             pr_err("unknown ioctl: cmd=%x ARGOIOCACCEPT=%lx\n", cmd,
```
[the remark this answers is not in the document]
Comment (nkrasnoff) [reply argo-linux/argo-module.c]: Good catch resolved

File: argo-linux/argo-module.c [hunk 4]
```diff
@@ -3380,100 +3380,142 @@ argo_ioctl(struct file *f, unsigned int cmd, unsigned long arg)
...
-
     if (_IOC_TYPE (cmd) != ARGO_TYPE)
         return rc;
 
     switch (cmd)
     {
         case ARGOIOCSETRINGSIZE:
+        {
+            uint32_t ring_size;
+            pr_debug("Start: SETRINGSIZE=%x pid=%d\n", cmd, current->pid);
```
[the remark this answers is not in the document]
Comment (nkrasnoff) [reply argo-linux/argo-module.c:3196]: I updated the commit message as you suggested but kept the `SETRINGSIZE` log message the same to keep with the convention to debug the commands inside the function that handles them. Changing this convention sounds like a good idea but I think should be handled separately from this PR. Thanks for the feedback!

**Comment by eric-ch** [comment]
The changes deferred are logged in: https://openxt.atlassian.net/browse/OXT-1795
Build [$10138](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10138)

File: argo-linux/argo-module.c [hunk 5]
```diff
@@ -3197,7 +3197,7 @@ argo_open_dgram(struct inode *inode, struct file *f)
     INIT_LIST_HEAD(&p->pending_recv_list);
     atomic_set(&p->pending_recv_count, 0);
 
-    pr_debug("argo_open dgram priv %p\n", p);
+    pr_debug("private: %p\n", p);
```
[the remark this answers is not in the document]
Comment (dozylynx) [reply argo-linux/argo-module.c:3005]: Change request for this filed by @eric-ch  here: https://openxt.atlassian.net/browse/OXT-1795 so will resolve this conversation now

File: argo-linux/argo-module.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (dozylynx) [reply argo-linux/argo-module.c:3196]: ok -- thanks!

**Comment by eric-ch** [comment]
Will merge soon.

**Pull request merged**